crates/autopilot/src/database/auction.rs

```diff
@@ -54,9 +72,15 @@ impl QuoteStoring for Postgres {
         let quote = database::quotes::find(&mut ex, &params)
             .await
             .context("failed finding quote by parameters")?;
-        quote
-            .map(|quote| Ok((quote.id, quote.try_into()?)))
-            .transpose()
+        if let Some(quote) = quote {
+            let quote_id = quote.id;
+            let quote_interactions = Self::get_quote_interactions(&mut ex, quote_id).await?;
+            let mut quote_data = QuoteData::try_from(quote)?;
+            quote_data.interactions = quote_interactions;
```

Comment:
This mutation can be avoided by using a query with JOIN. But I am still not sure if we need separate tables at all.